extension/vector/src/index/hnsw_index_utils.cpp

```diff
@@ -12,41 +12,62 @@
-    IndexOperation indexOperation) {
+    IndexOperation indexOperation, common::ConflictAction conflictAction) {
     auto transaction = transaction::Transaction::Get(context);
     switch (indexOperation) {
     case IndexOperation::CREATE: {
-        if (catalog::Catalog::Get(context)->containsIndex(transaction, tableEntry->getTableID(),
-                indexName)) {
-            throw common::BinderException{common::stringFormat(
-                "Index {} already exists in table {}.", indexName, tableEntry->getName())};
+        if (indexExists(context, transaction, tableEntry, indexName)) {
+            switch (conflictAction) {
+            case common::ConflictAction::ON_CONFLICT_THROW:
+                throw common::BinderException{common::stringFormat(
+                    "Index {} already exists in table {}.", indexName, tableEntry->getName())};
+            case common::ConflictAction::ON_CONFLICT_DO_NOTHING:
+                return true;
+            case common::ConflictAction::INVALID:
+                KU_UNREACHABLE;
+            }
         }
+        return false;
     } break;
     case IndexOperation::DROP:
     case IndexOperation::QUERY: {
-        if (!catalog::Catalog::Get(context)->containsIndex(transaction, tableEntry->getTableID(),
-                indexName)) {
-            throw common::BinderException{common::stringFormat(
-                "Table {} doesn't have an index with name {}.", tableEntry->getName(), indexName)};
+        if (!indexExists(context, transaction, tableEntry, indexName)) {
+            switch (conflictAction) {
+            case common::ConflictAction::ON_CONFLICT_THROW:
+                throw common::BinderException{
+                    common::stringFormat("Table {} doesn't have an index with name {}.",
+                        tableEntry->getName(), indexName)};
+            case common::ConflictAction::ON_CONFLICT_DO_NOTHING:
+                return false;
+            case common::ConflictAction::INVALID:
+                KU_UNREACHABLE;
+            }
         }
+        return true;
     } break;
     default: {
         KU_UNREACHABLE;
     }
     }
 }
 
-catalog::NodeTableCatalogEntry* HNSWIndexUtils::bindNodeTable(const main::ClientContext& context,
-    const std::string& tableName, const std::string& indexName, IndexOperation indexOperation) {
+catalog::TableCatalogEntry* HNSWIndexUtils::bindTable(const main::ClientContext& context,
+    const std::string& tableName) {
     binder::Binder::validateTableExistence(context, tableName);
     auto transaction = transaction::Transaction::Get(context);
     const auto tableEntry =
         catalog::Catalog::Get(context)->getTableCatalogEntry(transaction, tableName);
     binder::Binder::validateNodeTableType(tableEntry);
-    validateIndexExistence(context, tableEntry, indexName, indexOperation);
-    return tableEntry->ptrCast<catalog::NodeTableCatalogEntry>();
+    return tableEntry;
```

Comment:
You can still cast tableEntry as NodeTableCatalogEntry since you have validated the table type in line 69.